rfcs/0028-nix-release-model.md

```diff
@@ -0,0 +1,158 @@
+---
+feature: nix-release-model
+start-date: 2018-03-28
+author: Shea Levy <shea@shealevy.com>
+co-authors: Michael Raskin <7c6f434c@mail.ru> (any other Nix committers on-board?)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+A model of when, what, and how to make new releases of Nix.
+
+# Motivation
+[motivation]: #motivation
+
+We want to set clear expectations for users, provide guidance for
+developers, and provide both groups the benefits of easy availability
+of new improvements.
+
+# Detailed design
+[design]: #detailed-design
+
+The proposed process contains a number of interrelated components.
+I've numbered them for ease of reference and discussion, not to
+indicate any meaningful ordering. Any aspect of this can be suspended
+at the security team's discretion when handling a vulnerability.
+
+1. Before merging to `master`, require testing, documentation,
+   changelog entries, removal of WIP markers, etc. to keep `master`
+   somewhat stable, maintain bisectability, and shorten the release
+   preparation process. This includes both objective and subjective
+   components, and will always involve something of a judgment call.
+2. The Nix core team shall decide when a new release is warranted,
+   through its normal decision procedures. That decision will also
+   include designating a release manager for the release. The specific
+   conditions for a release being warranted are intentionally out of
+   scope of this RFC, but it is expected that the core team will
+   balance both the needs of the community at large and the needs of
+   developers working on Nix. In particular, if an especially major
```

Comment:
Not sure if the tradeoff is between developers and users; I would says that both groups can be split w.r.t. the real tradeoff: stability vs. development progress.

Reply:
Ah right, updated.